compiler/qsc_frontend/src/resolve.rs

```diff
@@ -930,30 +967,56 @@ impl Resolver {
                     }
                 }
                 let scope = self.current_scope_mut();
-                scope
-                    .tys
-                    .insert(local_name.clone(), ScopeItemEntry::new(id, item_source));
+                scope.tys.insert(
+                    local_name.clone(),
+                    ScopeItemEntry::new(id, item_source.clone()),
+                );
             }
 
             let res = match (term_result, ty_result) {
-                (Ok(res @ Res::Item(..)), _) | (_, Ok(res @ Res::Item(..))) => res,
+                // This is kind of a messy match, it is merged and formatted this way
```

Comment:
nit: I would place the "... is kind of a messy match ..." part of the comment before the `match` expression, leaving the explanation for the first case here. That way each relevant case gets its own comment :)